modules/e2ee/Context.js

```diff
@@ -250,17 +253,25 @@ export class Context {
 
             encodedFrame.data = newData;
         } catch (error) {
-            if (ratchetCount < RATCHET_WINDOW_SIZE) {
-                material = await importKey(await ratchet(material));
+            if (!this._externallyManaged) {
+                if (ratchetCount === 0) {
```

Comment:
Please leave a thick comment above this, explaining why we are doing it that way, but more importantly, this is not really related to the spirit of this PR, so let's make it its own commit at least.